ipaserver/install/dnskeysyncinstance.py

```diff
@@ -231,23 +305,17 @@ def __setup_softhsm(self):
             entropy_bits=0, special=None, min_len=pin_length)
 
         logger.debug("Saving user PIN to %s", paths.DNSSEC_SOFTHSM_PIN)
-        named_fd = open(paths.DNSSEC_SOFTHSM_PIN, 'w')
-        named_fd.seek(0)
-        named_fd.truncate(0)
-        named_fd.write(pin)
-        named_fd.close()
-        os.chmod(paths.DNSSEC_SOFTHSM_PIN, 0o770)
-        # chown to ods:named
-        os.chown(paths.DNSSEC_SOFTHSM_PIN, self.ods_uid, self.named_gid)
+        with open(paths.DNSSEC_SOFTHSM_PIN, 'w') as f:
+            # chown to ods:named
+            os.fchown(f.fileno(), self.ods_uid, self.named_gid)
+            os.fchmod(f.fileno(), 0o770)
```

Comment:
Why does this file have executable permission? The original code looks wrong. It's not an executable file.

```suggestion
            os.fchmod(f.fileno(), 0o660)
```